### PR TITLE
8341495: JShell crashes with java.util.MissingFormatArgumentException

### PR DESCRIPTION
**Pull request by archiecobbs**
JShell prints some of its output using `printf()` formatted strings. However, there are a few spots where a string `str` to be printed is passed directly as a format string like `format(str)` instead of indirectly like `format("%s", str)`, even though the string `str` contains unknown or arbitrary content outside of JShell's control. As a result, JShell can crash suddenly with a `MissingFormatArgumentException` or similar exception if the string `str` happens to contain a `%` format specifier, etc.

Please review this patch which attempts to fix the places where an unknown string is passed directly as a format string.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8341495](https://bugs.openjdk.org/browse/JDK-8341495): JShell crashes with java.util.MissingFormatArgumentException (**Bug** - P4)

### Reviewers
 * [Jan Lahoda](https://openjdk.org/census#jlahoda) (@lahodaj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21339/head:pull/21339` \
`$ git checkout pull/21339`

Update a local copy of the PR: \
`$ git checkout pull/21339` \
`$ git pull https://git.openjdk.org/jdk.git pull/21339/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21339`

View PR using the GUI difftool: \
`$ git pr show -t 21339`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21339.diff">https://git.openjdk.org/jdk/pull/21339.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21339#issuecomment-2392458007)
</details>

### CODE REVIEW

File: src/jdk.jshell/share/classes/jdk/internal/jshell/tool/JShellTool.java [hunk 1]
```diff
@@ -804,36 +805,36 @@ String getResourceString(String key) {
...
-     * @param s the string to prefix
-     * @param pre the string to prepend to each line
-     * @param post the string to append to each line (replacing newline)
-     * @return the pre/post-fixed and bracketed string
+     * @param s the format string to prefix
+     * @param pre the string to prepend to each line (printf safe)
+     * @param post the string to append to each line (replacing newline; printf safe)
+     * @return the pre/post-fixed and bracketed format string
      */
     String prefix(String s, String pre, String post) {
```
[the remark this answers is not in the document]
Comment (archiecobbs) [reply src/jdk.jshell/share/classes/jdk/internal/jshell/tool/JShellTool.java]: Yes, thanks. Fixed in 4e692b457db.

**Comment by liach** [comment]
@lahodaj Can you take a look at this fix?

**Comment by archiecobbs** [comment]
Still looking to get this bug fix reviewed hopefully in time for 24. Thanks.

**Comment by lahodaj** [comment]
Overall, looks good to me. But I would suggest to drop/avoid the changes in non-English localizing bundles. These are typically done by the localization team. (The changes may seem innocuous, but I am not clear if there isn't a risk of e.g. conflicts when merging their version with the mainline version.)

**Comment by archiecobbs** [comment]
> I would suggest to drop/avoid the changes in non-English localizing bundles

OK I will take your word for it - but first I just want to clarify that this will leave the main branch in a state where, in any language other than English, jshell will crash if you type `/help set prompt`.

Is that OK? Should we confirm with an i18n developer?

Thanks for reviewing!

**Comment by archiecobbs** [comment]
@lahodaj - anything else needed here? Thanks again for taking a look.

**Review by lahodaj — APPROVED** [review]
Looks good to me. Thanks!

**Pull request closed without merging**